### ISSUE TITLE
[🚀 Feature]: Exclude specific drivers from the selenium manager

### ISSUE THREAD

**Issue by joerg1985** [SeleniumHQ/selenium#11636]
### Feature and motivation

Allow specific drivers to be excluded from the download by the selenium-manager.
e.g. set the system property to a protected value to disable the download
`System.setProperty("webdriver.ie.driver", "n/a");`

### Usage example

I am running the standalone server in a offline environment and only the drivers needed are installed and set by system properties.

At the startup of the standalone server the selenium manager will try to download e.g. the internet explorer driver.
The startup is blocked until the connection timeout is reached and a warning is logged.

`10:12:22.669 WARN [DriverService.findExecutable] - Unable to obtain driver using Selenium Manager: Unsuccessful command executed: [C:\Users\my-user\AppData\Local\Temp\selenium-manager78222028412008605999502358382989\selenium-manager.exe, --driver, IEDriverServer]`

**Comment by titusfortner** [comment]
Would it work for you if we allowed you to set an environment variable that turned off downloads entirely?

**Comment by joerg1985** [comment]
@titusfortner sure, a command line flag would also be okay for me. I was thinking the way to opt-out singe browsers might be more generic for other usecases.

**Comment by AhmedElaraby14** [comment]
@bonigarcia @titusfortner I have this Issue too,
when i worked with WebDriverManager i used this working around to force it to use specified version in my local cashe
   EX: WebDriverManager.edgedriver().version("108.0.1462.46").setup();

But now I didn't know how to handle it.

**Comment by titusfortner** [comment]
The Grid is special because it tries to get drivers for everything that is available. Right now there's no option to avoid that. I think the right answer is to set an environment variable, which will work for all client bindings and grid.

On the client side, you can avoid Selenium Manager by setting the location of the driver directly: https://www.selenium.dev/documentation/webdriver/getting_started/install_drivers/?language=java#4-hard-coded-location

**Comment by krmahadevan** [comment]
@titusfortner 

> The Grid is special because it tries to get drivers for everything that is available. Right now there's no option to avoid that. 

I looked at this a bit in detail and here's what I have inferred:

* All the implementations of `org.openqa.selenium.WebDriverInfo` have a method named `boolean isAvailable()` which internally tries to build a concrete driver service (for e.g., `SafariDriverService` (or) `ChromeDriverService`).
* Based on the return type of `isAvailable()` the Grid's node decides if a particular `WebDriverInfo` is to be considered as supported by default (when the user does not specify any specific flavors via `--driver-implementation`) or if it should be ignored because it's not relevant for the platform (for e.g., safari is not relevant in windows)
* Once the above has been narrowed out, the `DriverService` checks if the **driver binary's location** has been specified via :
    * the JVM argument (for e.g., `webdriver.safari.driver`, `webdriver.edge.driver`) (or) 
    * if it can be found in the `PATH`.
* Only if the above fails is when we would be delegating back to `SeleniumManager` to download the binary.

**So to accomplish**

> I am running the standalone server in a offline environment and only the drivers needed are installed and set by system properties.

**One needs to do the following:**

* Specify ONLY the flavours of browsers that one needs via the flag `--driver-implementation` (for e.g., if I need only firefox to be supported, then I can run the jar as `java -jar selenium-server-4.8.0.jar standalone --driver-implementation "firefox"`
* Now one also needs to ensure that whatever browser flavours that one wanted, their corresponding driver binaries are either specified via 
    * the JVM arguments (or) 
    * are available in the path variable. For e.g., `java -Dwebdriver.gecko.driver=/Users/krishnanmahadevan/temp -jar selenium-server-4.8.0.jar standalone --driver-implementation "firefox"` (or)
    * specified via the configuration `toml` file as documented [here](https://www.selenium.dev/documentation/grid/configuration/toml_options/#configuring-and-customising-drivers)

In a nutshell, if a user just does the pre-requisites (i.e., ensuring that ONLY the desired browser flavors as specified at the start of the node and their corresponding driver binaries are made available in either `PATH` or via JVM arguments), then its as good as one using a flag to silence Selenium Manager.

**Comment by titusfortner** [comment]
So, it's possible for a user to specify which browsers to use for the node, or to turn off auto discovery. 

The ideal solution for Selenium will be tracked in #11639

**Comment by bonigarcia** [comment]
I don't understand the original request by @joerg1985. I mean, when specifying some path for `IEDriverServer.exe` using a system property, Selenium Manager should not be executed, right?

```
System.setProperty("webdriver.ie.driver", "c:/path/to/IEDriverServer.exe");
```

That should be enough to disable Selenium Manager downloads. At least in the bindings, if I am not wrong (not sure about Grid).

Regarding the comment by @AhmedElaraby14, Selenium Manager already allows to specify of a fixed driver version. Nevertheless, there is no way to pass that driver version from the bindings yet. We are currently discussing alternative ways to carry out this configuration (e.g., with a [configuration file](https://github.com/SeleniumHQ/selenium/issues/11689) or with [environment variables](https://github.com/SeleniumHQ/selenium/issues/11690)).

**Comment by diemol** [comment]
This was fixed by https://github.com/SeleniumHQ/selenium/commit/6cb3e2f83fcaf5487c7fdcb73a370ca557b19179
